### DISCUSSION TITLE
Plans for type-safety in template literals

### DISCUSSION THREAD

**Discussion by cbovis** [chakra-ui/panda#2065] [Q&A] [answered]
I'm currently assessing Panda CSS and if we were to adopt it then we'd been looking at adoption through the template literal syntax. From the docs I can see that one of the Caveats mentioned for template literals is a lack of autocomplete support. Presumably this also means a lack of type safety for tokens in template literals?

I'm interested to know whether there's any plans to improve the DX around the use of tokens in template literals or is this inherent to some kind of constraint in the design of Panda that's unlikely to be improved any time soon?

**Comment by segunadebayo** [comment, marked as answer]
Hi @cbovis,

Currently, we don't have plans for type safety in template literals. Our upcoming [ESLint plugin](https://github.com/chakra-ui/eslint-plugin-panda) might be able to help, though, so you get a hint when using invalid token paths.

**Reply by cbovis** [reply]
Great recommendation, thanks!

**Reply by anubra266** [reply]
Sneak peak @cbovis 
![image](https://github.com/chakra-ui/panda/assets/30869823/a1581fc7-ac82-48d3-bf24-92dd2624b0a7)
